packages/safe-core-sdk-types/src/types.ts

```diff
@@ -99,8 +97,7 @@ export interface BaseTransactionResult {
 }
 
 export interface TransactionResult extends BaseTransactionResult {
-  promiEvent?: PromiEvent<TransactionReceipt>
-  transactionResponse?: ContractTransactionResponse
+  transactionResponse: ContractTransactionResponse
```

Comment:
This could generate a breaking change later on if we go into Viem internally

Reply:
mmm i removed the promievent because is web3 specific. ContractTransactionResponse is ethers specific yes, do you mean we should use a internal type ?

Reply:
Hey i did several things here:
- I realized the `EthersTransactionOptions` after removing the `web3` specific stuff is duplicated with the `TransactionOptions`. Same for `EthersTransactionResult` so I removed them and updated a lot of generic types. 
- I renamed the `AdapterSpecificContractFunction` to `SafeContractFunction`
- Removed the ethers dependency from `safe-core-sdk-types` as it's no longer needed
- I set the `TransactionResponse` as unkwown but not completely sure about if this is the best option. It works without more changes at the SDK level

cc: @DaniSomoza @tmjssz